src/main/java/org/folio/dataexp/service/export/strategies/ld/ResourceUrlProvider.java

```diff
@@ -0,0 +1,23 @@
+package org.folio.dataexp.service.export.strategies.ld;
+
+import java.util.function.LongFunction;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provide a means to supply the host portion of a resource URL.
+ * This is not stored in the graph, and this class is named as it
+ * is so the export converter can find it and use it to generate URLs.
+ */
+@Service
+@RequiredArgsConstructor
+public class ResourceUrlProvider implements LongFunction<String> {
```

Comment:
Hi @zepheiryan , I can be wrong, but why do you need to implement a functional interface if the class is annotated with Spring @Service? It looks, this service will not be used as a lambda or a method reference. It is just a service that returns a URL. Because of that, implementing a  functional interface looks unnecessary here. Maybe we should not implement any interface at all, or create our own interface with appropriate method name?

Reply:
Hi @khandramai, this comes from the implementation in [lib-linked-data-rdf4ld](https://github.com/search?q=repo%3Afolio-org%2Flib-linked-data-rdf4ld%20resourceUrlProvider&type=code), which expects to find a bean of this name and signature. We'll ultimately replace this with something that communicates with mod-settings to retrieve the hostname that goes into assembling the return value, so I went ahead and made it a service in anticipation of doing so.

This is similar to [how it's done in mod-linked-data](https://github.com/folio-org/mod-linked-data/blob/master/src/main/java/org/folio/linked/data/service/rdf/ResourceUrlProvider.java), where a mod-configuration service is used to supply the hostname value.